Cython/Shadow.pyi

```diff
@@ -100,3 +103,19 @@ class typedef(CythonType, Generic[_T]):
 #    def __init__(self) -> None: ...
 
 #def fused_type(*args: Tuple[_T]) -> Type[FusedType[_T]]: ...
+
+class _EmptyDecoratorAndManager(object):
+    def __call__(self, x: _T) -> _T: ...
+
+    def __enter__(self) -> None: ...
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc: Optional[BaseException],
+        tb: Optional[TracebackType]
+    ) -> None: ...
+
+cclass = ccall = cfunc = _EmptyDecoratorAndManager()
```

Comment:
Idk, maybe it'll be better to refactor this into
```python
cclass: _EmptyDecoratorAndManager
...
```

Reply:
That won't work, but that does bring up something I wasn't sure about. This was copied from the Shadow.py file,

I don't completely understand why the following won't work:
```py
def cfunc(func: _T) -> _T:
    return func
```
`_EmptyDecoratorAndManager` implements the context manager protocol, but I've never seen them used like that:
```py
with cython.cfunc:
    ...
```

Reply:
It's more common to use things like boundscheck and wraparound as a with-block (to turn them off for a local bit of the function) but in principle most compiler directives can be used in a with-block.